science-platform/templates/telegraf-application.yaml

```diff
@@ -0,0 +1,29 @@
+{{- if .Values.telegraf.enabled -}}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: telegraf
+spec:
+  finalizers:
+    - kubernetes
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: telegraf
+  namespace: argocd
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io
+spec:
+  destination:
+    namespace: telegraf
+    server: https://kubernetes.default.svc
+  project: default
+  source:
+    path: services/telegraf
+    repoURL: {{ .Values.repoURL }}
+    targetRevision: {{ .Values.revision }}
+    helm:
+      valueFiles:
```

Comment:
Here I think you also need to list the `values.yam`l file which has the default telegraf configuration. It has to be first in the list followed by the environment-specific configuration so that they take precedence.